src/views/CoCurricularTranscript/index.tsx

```diff
@@ -20,30 +21,27 @@ const SectionTitle: { [Key in keyof TranscriptItems]: string } = {
 const CoCurricularTranscript = () => {
   const [loading, setLoading] = useState(true);
   const [transcriptItems, setTranscriptItems] = useState<TranscriptItems | undefined>();
-  const { profile, loading: loadingProfile } = useUser();
+  const isAuthenticated = useIsAuthenticated();
+  const { profile: userProfile, loading: loadingProfile } = useUser();
+  const profile = isAuthenticated ? userProfile : student1;
 
   useEffect(() => {
-    const loadTranscript = async () => {
-      if (!profile) return;
-
+    if (isAuthenticated) {
+      if (!profile) {
+        return;
+      }
       setLoading(true);
-
-      await transcriptService.getItems(profile.AD_Username).then(setTranscriptItems);
-
-      setLoading(false);
-    };
-
-    loadTranscript();
+      transcriptService.getItems(profile.AD_Username).then(setTranscriptItems);
```

Comment:
This actually won't work exactly as expected currently. Because you aren't `await`ing these `Promise`s, the browser will start the `getItems` request and then continue on before it's finished, executing `setLoading(false)` potentially **before** the `Promise` from the `getItems` request has resolved, meaning we haven't actually finished loading data yet.

To fix that, you can put the code inside the `useEffect` into an async function and invoke that function at the end of the `useEffect`.

Reply:
The best way to do this is to define the async function within the useEffect, and then call it immediately after it's defined.

It should look something like this:
```javascript
useEffect(() => {
  const loadTranscript = () => {
    if (isAuthenticated) {
      ...
      const transcriptItems = await transcriptSerivce.getItems();
      setTranscriptItems(transcriptItems);
    } else {
      setTranscriptItems(transcriptItems1);
    }
    setLoading(false);
  }
  loadTranscript();
}, [isAuthenticated, profile]);
```
or if you still prefer the `.then` syntax:
```javascript
useEffect(() => {
  const loadTranscript = () => {
    if (isAuthenticated) {
      ...
      await transcriptSerivce.getItems().then(setTranscriptItems);
    } else {
      setTranscriptItems(transcriptItems1);
    }
    setLoading(false);
  }
  loadTranscript();
}, [isAuthenticated, profile]);
```

Reply:
Sounds good lol! It works